api/tests/routes/adage_iframe/post_collective_request_test.py

```diff
@@ -0,0 +1,143 @@
+from datetime import datetime
+import logging
+
+import pytest
+
+from pcapi.core.educational import factories as educational_factories
+from pcapi.routes.adage_iframe.serialization import offers
+
+from tests.conftest import TestClient
+from tests.routes.adage_iframe.utils_create_test_token import create_adage_valid_token_with_email
+
+
+pytestmark = pytest.mark.usefixtures("db_session")
+
+stock_date = datetime(2021, 5, 15)
+educational_year_dates = {"start": datetime(2020, 9, 1), "end": datetime(2021, 8, 31)}
+
+
+class Returns200Test:
+    def test_post_collective_request(self, app, caplog, client):
+        educational_institution = educational_factories.EducationalInstitutionFactory()
+        educational_redactor = educational_factories.EducationalRedactorFactory(email="JamesHolden@rocinante.com")
+        offer = educational_factories.CollectiveOfferTemplateFactory()
+        offer_id = offer.id
+
+        adage_jwt_fake_valid_token = create_adage_valid_token_with_email(
+            email=educational_redactor.email, uai=educational_institution.institutionId
+        )
+        test_client = TestClient(app.test_client())
+        test_client.auth_header = {"Authorization": f"Bearer {adage_jwt_fake_valid_token}"}
```

Comment:
`TestClient` ne devrait plus être utilisé. La fixture `client` est équivalente.

```suggestion
        adage_jwt_fake_valid_token = create_adage_valid_token_with_email(
            email=educational_redactor.email, uai=educational_institution.institutionId
        )
        client.auth_header = {"Authorization": f"Bearer {adage_jwt_fake_valid_token}"}
```

D'autre part, il serait assez agréable à l'oeil de faire une fonction utilitaire `with_adage_token()` qui permette de simplifier ce bloc en : 

```suggestion
        client = client.with_adage_token(educational_redactor, educational_institution)
```

Je vois ces mêmes lignes dans beaucoup de tests sur les offres collectives, c'est bien plus verbeux que nécessaire.

Reply:
quelque chose doit m'echaper mais les tests ne fonctionne pas lorsque je les passes ainsi.

Reply:
Quelle est l'erreur ?

Reply:
`tests/routes/adage_iframe/post_collective_request_test.py::Returns200Test::test_post_collective_request - AttributeError: 'TestClient' object has no attribute 'with_adage_token'. Did you mean: 'with_eac_token'?`

j'ai bien sur essayé avec with_eac_token mais il n'accepte pas les arguments une fois fait ainsi

`FAILED tests/routes/adage_iframe/post_collective_request_test.py::Returns200Test::test_post_collective_request - TypeError: TestClient.with_eac_token() takes 1 positional argument but 3 were given`

Reply:
`TestClient.with_adage_token()` n'existe pas. J'ai proposé que tu _ajoutes_ cette méthode.